mantidproject/mantid
# Fix deprecation warning and simplify Indirect Data Analysis external plotting

**Description of work**
This PR fixes a deprecation warning seen when building with the Ninja generator on Windows from a git bash terminal. I discovered that a large chunk of code is no longer used, including the `QtLazyAsyncRunner` class, and so it is nice that this can be removed. It simplifies the code greatly whilst also keeping the exact same functionality.

**To test:**

1. Open `Indirect`->`Data Analysis`
2. Go to ConvFit
3. Click Add workspace and load in the first `0-2` spectra for an `_red` and `_res` workspace in the sample dataset
4. Click Add and close the dialog
5. Select 2 Lorentzians
6. Click the `Plot Current Preview` button. It should produce an external plot
7. Tick the `Plot Guess` option. This should only show the guess on the internal plot. This is the same behaviour as previously (before this PR).

Fixes #36271

*This does not require release notes* because **there has been no change in functionality**

#### Reviewer ####

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

#### Gatekeeper ####

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
